handler/src/main/java/io/netty/handler/ssl/OpenSslServerContext.java

```diff
@@ -259,59 +336,64 @@ public OpenSslServerContext(
                 SSLContext.setVerify(ctx, SSL.SSL_CVERIFY_NONE, VERIFY_DEPTH);
 
                 /* Load the certificate chain. We must skip the first cert when server mode */
-                if (!SSLContext.setCertificateChainFile(ctx, certChainFile.getPath(), true)) {
+                if (!SSLContext.setCertificateChainFile(ctx, keyCertChainFile.getPath(), true)) {
                     long error = SSL.getLastErrorNumber();
                     if (OpenSsl.isError(error)) {
                         String err = SSL.getErrorString(error);
                         throw new SSLException(
-                                "failed to set certificate chain: " + certChainFile + " (" + err + ')');
+                                "failed to set certificate chain: " + keyCertChainFile + " (" + err + ')');
                     }
                 }
 
                 /* Load the certificate file and private key. */
                 try {
                     if (!SSLContext.setCertificate(
-                            ctx, certChainFile.getPath(), keyFile.getPath(), keyPassword, SSL.SSL_AIDX_RSA)) {
+                            ctx, keyCertChainFile.getPath(), keyFile.getPath(), keyPassword, SSL.SSL_AIDX_RSA)) {
```

Comment:
and maybe others like: SSL_AIDX_EC, #define SSL_AIDX_GOST94, and SSL_AIDX_GOST01